### PR TITLE
[EPMEDU-1510]: Approve feeding from the feeding card

### PR DESCRIPTION
**Pull request by Fileek**
### Ticket reference
https://jira.epam.com/jira/browse/EPMEDU-1510

### Description
 - Implemented approval action
 - Added shimmer loading for photos on feeding card

### Screenshot/Video
<details>
  <summary>Click to open</summary>

https://github.com/AnimealProject/animeal_android/assets/83027107/222ccd17-ca88-4c0e-bd90-ac789b4f39cc
</details>

### CODE REVIEW

File: shared/feature/feedings/src/main/java/com/epmedu/animeal/feedings/presentation/viewmodel/FeedingsState.kt [hunk 1]
```diff
@@ -9,5 +9,8 @@ data class FeedingsState(
     val feedingsFiltered: ImmutableList<FeedingModel> = persistentListOf(),
     val isLoading: Boolean = false,
```
Comment (mskibin) [inline shared/feature/feedings/src/main/java/com/epmedu/animeal/feedings/presentation/viewmodel/FeedingsState.kt]: Seems that `isLoading` property is not used anymore. Can we delete it?

File: shared/feature/feedings/src/main/java/com/epmedu/animeal/feedings/presentation/viewmodel/FeedingsViewModel.kt [hunk 2]
```diff
@@ -150,4 +171,33 @@ internal class FeedingsViewModel @Inject constructor(
...
     }
+
+    private fun approveFeeding() {
+        state.currentFeeding?.let { feeding ->
+            updateState { copy(isLockedAndLoading = true) }
+
+            viewModelScope.launch {
+                performAction(
+                    action = {
+                        approveFeedingUseCase(feeding.id, "")
```
Comment (mskibin) [inline shared/feature/feedings/src/main/java/com/epmedu/animeal/feedings/presentation/viewmodel/FeedingsViewModel.kt]: Will we have the possibility to enter the reason in the future? Seems there is no such functionality for now.

File: shared/feature/feedings/src/main/java/com/epmedu/animeal/feedings/presentation/viewmodel/FeedingsState.kt (hunk 1 above)
Comment (Fileek) [reply shared/feature/feedings/src/main/java/com/epmedu/animeal/feedings/presentation/viewmodel/FeedingsState.kt]: Good catch! I missed it during rebasing, will remove it. Thanks!

File: shared/feature/feedings/src/main/java/com/epmedu/animeal/feedings/presentation/viewmodel/FeedingsViewModel.kt (hunk 2 above)
Comment (Fileek) [reply shared/feature/feedings/src/main/java/com/epmedu/animeal/feedings/presentation/viewmodel/FeedingsViewModel.kt]: I have no clue, but we agreed with the backend team to pass `The request includes all necessary details.` for now.

Comment (Fileek) [reply shared/feature/feedings/src/main/java/com/epmedu/animeal/feedings/presentation/viewmodel/FeedingsViewModel.kt]: Updated.

File: shared/feature/feedings/src/main/java/com/epmedu/animeal/feedings/presentation/viewmodel/FeedingsState.kt (hunk 1 above)
Comment (Fileek) [reply shared/feature/feedings/src/main/java/com/epmedu/animeal/feedings/presentation/viewmodel/FeedingsState.kt]: Removed.

**Review by mskibin — APPROVED** [review]

**Pull request merged**